Added 2024 update for newly acquired short code (#55)
cds-snc/notification-adr: records/2021-09-20.scalibility.notify-short-code.md

- **One short code for closely related use cases**: It is highly unlikely that telco carriers will approve one single short code  for all federal programs. Ideally, user cases sharing same short code are logically related, connected and target same subscriber-base. For example, ArriveCAN and Travel Advisory messages for travelers. It is recommended to highlight commonality of use cases sharing same short code in the application

- **It is recommended to include the following details in the application document**: Detailed description of opt-in process for each use case: screenshots, approval process, enforcement process SMS message templates: constant part  and variable parts of messages who does control and  fill-in the template for each use case (GCNotify team or GoC departments). This information is required by carriers to understand who is a contact person/team if quick updates is needed.

Given that there is a strong preference for one short code only for closely related use cases, and given that it is recommended to include detailed documentation for each use case, the financial and administrative costs at this time are too high for us to proceed with our application.

> We will reconsider opening our file for a short code, or multiple shortcode if the need arises and the use case is a good fit for the program and parameters given by AWS, CWTA and the carriers.

Author: ```suggestion
We will reconsider opening our file for a short code, or multiple shortcode if the need arises and the use case is a good fit for the program and parameters given by AWS, CTA and the carriers.
```

## Update 2024

We revived the effort in 2023 due to our scaling need for SMS, coming as a requirement for getting provincial beta onboarding program within GCNotify.

> A policy research was performed to dig into CASL legislation and see if that should apply to governmental bodies. The findings were that it should not. Considering this and after talks with AWS (who talked to aggregators), it was deemeed that it would be a safe strategy to go forward with a specialized use case for a new short code request. This would get us a foot in the door, while looking for options to upgrade our existing short code to a generic usage once we got it by trying to get in touch with the CWTA organization.

Author: ```suggestion
A policy research was performed to dig into CASL legislation and see if that should apply to governmental bodies. The findings were that it should not. Considering this and after talks with AWS (who talked to aggregators), it was deemeed that it would be a safe strategy to go forward with a specialized use case for a new short code request. This would get us a foot in the door, while looking for options to upgrade our existing short code to a generic usage once we got it by trying to get in touch with the CTA organization.
```

Hence, we decided to go forward with a short code specialized for 2FA use case, considering this is one of our most common use case among GCNotify's users and it makes up 50% of our SMS notification traffic. Working closely with product and the policy team, we filled out a request that provided our governmental policies and using example of GCNotify itself as a 2FA example.

> In the meantime, we approched the CWTA organization and knocked at their door. We met with one of their representative to listen to our use case for a generic usage of our newly acquired short code that we recently acquired a week before (good timing!). At the time of this writing, we are in communication and hope we can get through to have a broader usage short code but nothing is guaranteed. This is a continuous and patient effort.

Author: ```suggestion
In the meantime, we approached the CTA organization and knocked at their door. We met with one of their representative to listen to our use case for a generic usage of our newly acquired short code that we recently acquired a week before (good timing!). At the time of this writing, we are in communication and hope we can get through to have a broader usage short code but nothing is guaranteed. This is a continuous and patient effort.
```

Ideally, our next steps would have us the upgraded generic usage short code, with a 3 letter code.

## Consequences

- On January 16th, 2024, we acquired our new short code.
- With our newly acquired short code (in 2024), we are able to send 100 SMS per second. This provides a potential of 6,000 SMS notifications per minute.

> - This is limited to 2FA authentication usage though and these are less likely to see outburst of notifications which would max out the short code sending rate capacity.

Author: ```suggestion
- This is limited to 2FA authentication usage though and these are less likely to see burst of notifications which would max out the short code sending rate capacity.
```
